cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
I agree: this is the biggest question in my head as well. 

In this case, I don't think the issue is the use of hostPort: I added a "hostNetwork" clause, not a "hostPort" clause. 

To be clear, I understand why it was slow. It was slow because the pod couldn't access the parent DNS because it didn't have connectivity to the node's network. This change allows us to access the parent DNS because we have access to the node's network.

I think the possible resolutions are:

- Accept as-is because it's a reasonable change
- Disable inheriting DNS when using Photon Controller... but then when pods lookup DNS records, they can only find the ones in the Kubernetes cluster. (Will users expect to inherit DNS?)
- Figure out how to have the pod access the node's network to access DNS. (Is this appropriate?)
- ... Something else I don't know about.

If it helps to discuss this live (phone, Hangout, Slack), I'm happy to do that. I'm free all day except 2:00 - 3:00pm Pacific.